ci-operator/config/stolostron/kube-state-metrics/stolostron-kube-state-metrics-release-2.14.yaml

```diff
@@ -11,6 +11,10 @@ base_images:
     name: builder
     namespace: stolostron
     tag: go1.22-linux
+  stolostron_builder_go1.23-linux:
```

Comment:
Is there are reason for having multiple go builders ? Seems the `image_stream_tag` builder uses 1.22 whereas the `Dockerfile.prow` image is built with 1.23. I am not sure exactly what the former build does but would prefer if we could get rid of everything but 1.23 so we are consistent.

Reply:
No reason to keep multiple go builders, afaik @jacobbaungard . I removed older entries, also updated `image_stream_tag` to point to 1.23.

Reply:
/pj-rehearse pull-ci-stolostron-prometheus-operator-release-2.14-images